scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +5216,128 @@ def __rtruediv__(self, other):
 
     def __neg__(self):
         return self * -1
+
+
+class IMFTransformedDistribution(TransformedDistribution):
+    r"""Distribution underlying an injective, monotonic function of a random variable
+
+    Given a random variable :math:`X`; an injective, monotonic function
+    :math:`g(u)`, its inverse :math:`h(u) = g^{-1}(u)`, the derivative
+    :math:`h'(u) = \frac{dh(u)}{du}`, define the distribution underlying
+    the random variable :math:`Y = g(X)`.
+
+    Parameters
+    ----------
+    X : `ContinuousDistribution`
+        The random variable :math:`X`.
+    g, h, dh : callable
+        Elementwise functions representing the mathematical functions
+        :math:`g(u)`, :math:`h(u)`, and :math:`h'(u)`
+    logdh : callable, optional
+        Elementwise function representing :math:`\log(h'(u))`.
+        The default is ``lambda u: np.log(dh(u))``, but providing
+        a custom implementation may avoid over/underflow.
+
+    """
+
+    def __init__(self, dist, *args, g, h, dh, logdh=None, **kwargs):
+        super().__init__(dist, *args, **kwargs)
+        self._g = g
+        self._h = h
+        self._dh = dh
+        self._logdh = (logdh if logdh is not None
+                       else lambda u: np.log(dh(u)))
+
+    def _overrides(self, method_name):
+        # Do not use the generic overrides of TransformedDistribution
+        return False
+
+    def _support(self, **params):
+        # Add shortcut for infinite support?
+        a, b = self._dist._support(**params)
+        return self._g(a), self._g(b)
+
+    def _logpdf_dispatch(self, x, *args, **params):
```

Comment:
```suggestion
    def _logpdf_formula(self, x, *args, **params):
```
Something to consider in the future for all transformed distributions - don't override the `_dispatch` function, just the `_formula`.

Reply:
This makes sense, yeah.